schedule.py

```diff
@@ -0,0 +1,54 @@
+from typing import Iterable
+from flask_restful import abort
+
+from scheduledao import ScheduleDAO
+from event import Event
+
+class Schedule:
+    dao = ScheduleDAO()
+
+    def __init__(self, owner: str, events: Iterable[Event]=[]):
+        self.owner = owner
+        self.events = set(events)
+
+    @classmethod
+    def fromdict(cls, sched_dict):
+        owner = sched_dict['owner']
+        events = [Event.get(e) for e in sched_dict['events']]
+        return cls(owner, events)
+
+    def add_event(self, event):
+        self.events.add(event)
+
+    def remove_event(self, event_name):
```

Comment:
this can be simplified, all it is doing is removing an event.. simply `del` it from the list instead of iterating over the entire list every time

Reply:
The function takes an event name as a parameter, so I can't access an element directly to `del` it. I could just `del` it if I did a `GET` operation with the name to get the event, but I feel like adding a db operation is suboptimal, as there will likely not be many events in the list anyways.

Reply:
Each event should have an event ID .. this id will than be removed from the set, the set will actually only ever hold a list if ID's .. this abstracts the concept of having a list of one object stored in the database of another object .. what happens when we edit that event, does the schedule get updated automatically ?